Fix AMD GPUs not being detected

max-maag opened the pull request:
## Summary

Each version of torch is only available for specific versions of CUDA and ROCm. The Invoke installer tries to install torch 2.4.1 with ROCm 5.6 support, which does not exist. As a result, the installation falls back to the default CUDA version so AMD GPUs aren't detected. This commits fixes that by bumping the ROCm version to 6.1, as suggested by the PyTorch documentation.[^1] Torch 2.4.1 does not appear to be available for ROCm 6.2.

The specified CUDA version of 12.4 is still correct according to [^1] so it does need to be changed.

[^1]: https://pytorch.org/get-started/previous-versions/#v241

## Related Issues / Discussions

Closes #7006
Closes #7146

## QA Instructions

- Install Invoke 5.1.1 or later with ROCm support using the installer on a system with an AMD GPU.
- Start the server.
- Generate any image.

Without this fix, the  CPU is used to generate images. This can be seen in the log output. Image generation also takes forever.

I did not test the changes to the Dockerfile since I am not familiar with Docker.

## Merge Plan

n/a

## Checklist

- [ ] _The PR has a short but descriptive title, suitable for a changelog_
- [ ] _Tests added / updated (if applicable)_
- [ ] _Documentation added / updated (if applicable)_

ebr commented:
We don't have any way of testing this on a 6xxx architecture, but this seems like a worthwhile and correct change nonetheless. We do know that 7xxx (Navi32/33 / RDNA3) chips will need some more special handling, but that shouldn't block this update.

@max-maag - we also have this index url set in the Dockerfile - could you please update that in this PR?
Thanks for your contribution!

max-maag commented:
>we also have this index url set in the Dockerfile - could you please update that in this PR? Thanks for your contribution!

I saw that someone reported that in #7006 but because I'm not using Docker and am not familiar with it at all I kept it out of this PR until now. I changed the URL in the Docker file and added the relevant issue to this PR's related issue list.

ebr commented:
Approved - LGTM from my perspective. Thanks again for the contribution.

Just FYI - I finally got it to generate on a recent AMD GPU (W7900). Here's a full write-up: https://gist.github.com/ebr/e4e4118b603bd95bfd2408ee30c27f0a. It's not pretty, but it works.
